### PR TITLE
Enable resetting template instances

### PR DESCRIPTION
**Pull request by thabetx**
Added a context menu item to reset template instances.

### CODE REVIEW

File: src/tiled/abstractobjecttool.cpp [hunk 1]
```diff
@@ -57,6 +57,12 @@ static bool isResizedTileObject(MapObject *mapObject)
     return false;
 }
 
+static bool isChangedTemplateInstance(MapObject *mapObject){
```
Comment (bjorn) [inline src/tiled/abstractobjecttool.cpp]: Coding style: please put `{` on its own line for functions.

File: src/tiled/changemapobject.cpp [hunk 2]
```diff
@@ -203,3 +203,63 @@ void DetachObjects::undo()
...
+                               QUndoCommand *parent)
+    : QUndoCommand(QCoreApplication::translate("Undo Commands",
+                                               "Reset %n Instnaces",
+                                               nullptr, mapObjects.size()), parent)
+    , mMapDocument(mapDocument)
+    , mMapObjects(mapObjects)
+{
+    for (const MapObject *object : mMapObjects) {
+        mProperties.append(object->properties());
+        mChangedProperties.append(object->changedProperties());
```
Comment (bjorn) [inline src/tiled/changemapobject.cpp]: `mChangedProperties` isn't actually used.

File: src/tiled/changemapobject.cpp [hunk 3]
```diff
@@ -203,3 +203,63 @@ void DetachObjects::undo()
...
+
+    emit mMapDocument->objectsChanged(mMapObjects);
+
+    // This signal forces updating custom properties in the properties dock
+    emit mMapDocument->selectedObjectsChanged();
+}
+
+void ResetInstances::undo()
+{
+    // TODO: Is there a better way than saving clones and resetting like this?
```
Comment (bjorn) [inline src/tiled/changemapobject.cpp]: Maybe, but I think this is fine.

Comment (thabetx) [reply src/tiled/changemapobject.cpp]: Alright, I found out that I didn't update name or type, I will add them.

File: src/tiled/changemapobject.cpp (hunk 2 above)
Comment (thabetx) [reply src/tiled/changemapobject.cpp]: Okay, even `mProperties` isn't used, I didn't think I will clone the objects when I created them.

File: src/tiled/changemapobject.cpp (hunk 3 above)
Comment (thabetx) [reply src/tiled/changemapobject.cpp]: Also updating the `templateRef` was unnecessary.

**Comment by thabetx** [comment]
I've updated the PR.

**Pull request merged**